lib/src/gitignore.rs

```diff
@@ -306,37 +208,63 @@ mod tests {
     fn test_gitignore_unusual_symbols() {
         assert!(matches(b"\\*\n", "*"));
         assert!(!matches(b"\\*\n", "foo"));
-        assert!(matches(b"\\\n", "\\"));
         assert!(matches(b"\\!\n", "!"));
         assert!(matches(b"\\?\n", "?"));
         assert!(!matches(b"\\?\n", "x"));
+        assert!(matches(b"\\w\n", "w"));
+        assert!(GitIgnoreFile::empty().chain("", b"\\\n").is_err());
+    }
+
+    #[test]
+    #[cfg(not(target_os = "windows"))]
+    fn teest_gitignore_backslash_path() {
+        assert!(!matches(b"/foo/bar", "/foo\\bar"));
+        assert!(!matches(b"/foo/bar", "/foo/bar\\"));
+
+        assert!(!matches(b"/foo/bar/", "/foo\\bar/"));
+        assert!(!matches(b"/foo/bar/", "/foo\\bar\\/"));
+
         // Invalid escapes are treated like literal backslashes
+        assert!(!matches(b"\\w\n", "\\w"));
+        assert!(matches(b"\\\\ \n", "\\ "));
+        assert!(matches(b"\\\\\\ \n", "\\ "));
+    }
+
+    #[test]
+    #[cfg(target_os = "windows")]
+    /// ignore crate consider backslaes as a directory divider onln on Windows.
```

Comment:
```suggestion
    /// ignore crate consider backslashes as a directory divider only on Windows.
````

Does this mean we also need to check for `\` at the end of a path on Windows, instead of only `/`?

Reply:
I don't think so. path already provided after [`to_internal_dir_string`](https://github.com/martinvonz/jj/blob/ce295f8bc2a30a7baaffea0b4ad9fd2707357522/lib/src/repo_path.rs#L277) calling. it doesn't check OS.
 ```rust
pub fn to_internal_dir_string(&self) -> String {
        if self.value.is_empty() {
            String::new()
        } else {
            [&self.value, "/"].concat()
        }
    }
```